### DISCUSSION TITLE
Dex - Multiple Connectors workflow Issue

### DISCUSSION THREAD

**Discussion by SaurabhAZip** [dexidp/dex#3491] [Q&A]
I am using multiple Identity Providers say 
1. google and 
2. github

Now openid configuration is served through following URL
`https://DOMAIN_NAME/.well-known/openid-configuration`

and `https://DOMAIN_NAME` or  `https://DOMAIN_NAME/auth` opens up the page to choose between the 2 Idp's configured. 
<img width="322" alt="Screenshot 2024-04-23 at 3 54 49 PM" src="https://github.com/dexidp/dex/assets/108371014/bc4ac15c-f09a-415d-8402-3177648bc8ec">

Since, client uses `https://DOMAIN_NAME/.well-known/openid-configuration` path for auth, users have to select one of the IdP's configured (client uses root path for auth as configuration is served by root path)

However, I want client to use(for this scenario) `https://DOMAIN_NAME/auth/github` or `https://DOMAIN_NAME/auth/google `directly and pre-select one of the options.

For that to work, Dex should be able to serve 
`https://DOMAIN_NAME/auth/{connector}/.well-known/openid-configuration` so that client can directly call `https://DOMAIN_NAME/auth/{connector} for auth`. 

It might make sense to support `https://DOMAIN_NAME/auth/{connector}/.well-known/openid-configuration` end-point in case of multiple connectors configured in Dex.

**Comment by cardoe** [comment]
Unfortunately that won't work everything before the `/.well-known` is the `iss` or the issuer URL and must match for OIDC.
